linkerd/app/outbound/src/tls/concrete.rs

```diff
@@ -0,0 +1,366 @@
+use crate::{metrics::BalancerMetricsParams, stack_labels, BackendRef, Outbound, ParentRef};
+use linkerd_app_core::{
+    config::QueueConfig,
+    drain, io,
+    metrics::{
+        self,
+        prom::{self, EncodeLabelSetMut},
+    },
+    proxy::{
+        api_resolve::{ConcreteAddr, Metadata},
+        core::Resolve,
+        http::AuthorityOverride,
+        tcp::{self, balance},
+    },
+    svc::{self, layer::Layer},
+    tls::{self, ServerName},
+    transport::{self, addrs::*},
+    transport_header::SessionProtocol,
+    Error, Infallible, NameAddr,
+};
+use std::{fmt::Debug, net::SocketAddr, sync::Arc};
+use tracing::info_span;
+
+/// Parameter configuring dispatcher behavior.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum Dispatch {
+    Balance(NameAddr, balance::EwmaConfig),
+    Forward(Remote<ServerAddr>, Metadata),
+    Fail { message: Arc<str> },
+}
+
+/// A backend dispatcher explicitly fails all requests.
```

Comment:
```suggestion
/// A backend dispatcher that explicitly fails all requests.
```

Reply:
that suggestion fixes a typo, i believe, but is this comment affixed to the wrong structure? it seems like this is _an_ error returned by the `fail` service created below in this statement:

```rust
          let fail = svc::ArcNewService::new(|message: Arc<str>| {
                svc::mk(move |_| futures::future::ready(Err(DispatcherFailed(message.clone()))))
            });
```